### PR TITLE
feat: add ability for service name override

### LINKED ISSUE #435: Implement -n / -name flag
to set the service name. Each DS implementation will continue to have a hardcoded default name but this flag will override it.
This matches the C device services.

### PR DESCRIPTION
**Pull request by hahattan**
Fix #435 
pending on edgexfoundry/app-functions-sdk-go#356

Signed-off-by: Chris Hung <chris@iotechsys.com>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://wiki.edgexfoundry.org/display/FA/Committing+Code+Guidelines#CommittingCodeGuidelines-CommitMessages
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] Other... Please describe:

### CODE REVIEW

**Comment by lenny-intel** [comment]
> Prefer --name / EDGEX_SERVICE_NAME as that is what is used in EdgeX (metadata).

Ok, so device SDK is using this name for the Service Name stored in the DB as we'll as for the service key used in Consul. App SDK is only using it for the service key...

File: pkg/service/main.go [hunk 1]
```diff
@@ -25,9 +26,17 @@ import (
 	"github.com/gorilla/mux"
 )
 
+var serviceKeyOverride string
+
 func Main(serviceName string, serviceVersion string, proto interface{}, ctx context.Context, cancel context.CancelFunc, router *mux.Router, readyStream chan<- bool) {
 	startupTimer := startup.NewStartUpTimer(common.BootRetrySecondsDefault, common.BootTimeoutSecondsDefault)
 
+	sdkFlags := flags.New()
```
Comment (lenny-intel) [inline pkg/service/main.go:34]: Need to add the usesage for you addition command-line option. Here is a sample from App Funcs SDK:

```
	additionalUsage :=
		"    -s/--skipVersionCheck           Indicates the service should skip the Core Service's version compatibility check.\n" +
			"    -sk/--serviceKey                Overrides the service key to be used with Registry and/or Configuration Providers\n" +
			"                                    If the name provided contains the text `<profile>`, this text will be replaced with\n" +
			"                                    the name of the profile used."

	sdkFlags := flags.NewWithUsage(additionalUsage)

```

File: pkg/service/main.go [hunk 2]
```diff
@@ -25,9 +26,17 @@ import (
...
+	sdkFlags.FlagSet.StringVar(&serviceKeyOverride, "name", "", "")
```
Comment (lenny-intel) [inline pkg/service/main.go:35]: change `serviceKeyOverride` to `serviceNameOveride`

File: pkg/service/main.go [hunk 3]
```diff
@@ -25,9 +26,17 @@ import (
...
+	sdkFlags.FlagSet.StringVar(&serviceKeyOverride, "n", "", "")
+	sdkFlags.Parse(os.Args[1:])
+
+	serviceName = setServiceKey(serviceName, sdkFlags.Profile())
```
Comment (lenny-intel) [inline pkg/service/main.go:39]: Change `setServiceKey` to `setServiceName`

**Comment by lenny-intel** [comment]
Also not sure about the implementation of <profile> place holder in the name since this is not done in the C SDK. @iain-anderson , thoughts?

**Comment by iain-anderson** [comment]
> Also not sure about the implementation of place holder in the name since this is not done in the C SDK. @iain-anderson , thoughts?

No, that doesn't look right. The CSDK uses "name" if you don't set a profile, and "name;profile" if you do - is that still what is expected?

**Comment by lenny-intel** [comment]
> > Also not sure about the implementation of place holder in the name since this is not done in the C SDK. @iain-anderson , thoughts?
> 
> No, that doesn't look right. The CSDK uses "name" if you don't set a profile, and "name;profile" if you do - is that still what is expected?

Right, at this point I don't thin you want to have this `<profile>` place holder capability since it isn't in the CSDK. @hahattan

**Comment by cloudxxx8** [comment]
@iain-anderson please help review again, thanks

**Comment by iain-anderson** [comment]
app-functions-sdk have used -n / --serviceName, do the same here? otherwise lgtm

**Comment by hahattan** [comment]
changed to use `-n / --serviceName` to be consistent with app-functions-sdk

**Comment by cloudxxx8** [comment]
Please add a section into the Readme to introduce all the flags, especially the one in this PR

**Pull request merged**